metaflow/plugins/kfp/kfp.py

```diff
@@ -386,20 +354,27 @@ def build_kfp_component(node: DAGNode, task_id: str) -> KfpComponent:
 
             user_code_retries, total_retries = KubeflowPipelines._get_retries(node)
 
-            step_cli = self._step_cli(node, task_id, user_code_retries)
             resource_requirements = self._get_resource_requirements(node)
 
             return KfpComponent(
                 name=node.name,
-                cmd_template=self._command(
-                    self.code_package_url,
-                    self.environment,
-                    node.name,
-                    [step_cli],
-                    resource_requirements,
-                    task_id,
-                ),
+                init_cmd=self._init_command(self.code_package_url, self.environment),
+                cd_into_metaflow_package_cmd=self._cd_into_metaflow_package_cmd(),
```

Comment:
```python
                init_cmd=self._init_command(self.code_package_url, self.environment),
                cd_into_metaflow_package_cmd=self._cd_into_metaflow_package_cmd(),
```
These can  combined?

Reply:
I've merged `_init_command()` and `_cd_into_metaflow_package_cmd()` into a single function, `_get_metaflow_bootstrap_cmd()`, which produces two values: `_compile_time_bootstrap_cmd` and `_run_time_bootstrap_cmd`. `_compile_time_bootstrap_cmd` and `_run_time_bootstrap_cmd` are the same for all steps in a flow, and thus do not need to be part of a `kfp_component`. I instead instantiated them as member variable of the `KubeflowPipelines` class. This speeds up the compilation process.